### ISSUE TITLE
Highlight of identifiers broken for tab-indented Scala 3 code

### ISSUE THREAD

**Issue by mirzov** [scalameta/metals#3724]
### Describe the bug

The bug has been observed in VSCode, but since it occurs only with Scala 3, not with Scala 2, I'm assuming the problem is with metals. When putting a cursor on an identifier (e.g. variable or a method name), all the instances of the identifier get highlighted. The problem with tab-indented code is that the highlighted areas are off (see the screenshot).

Interestingly, fixing seemingly-related #3698 did not automatically solve this problem. (By the way, thanks for fixing it so promptly! The autocomplete problem is gone when using the latest snapshot of metals).

![scala_vscode_broken_highlight](https://user-images.githubusercontent.com/666648/158559982-98d8f9e6-129c-4b27-85e2-a3f7ed9a1d61.png)

### Expected behavior

The highlighted areas are expected to coincide with the occurrences of the identifier that the cursor has been placed on.

### Operating system

Linux

### Editor/Extension

VS Code

### Version of Metals

0.11.2+49-ee7372d7-SNAPSHOT

**Comment by tgodzik** [comment]
Thanks for reporting!

It looks like maybe `cursorPos.source.offsetToLine(offset)` is broken in the Dotty compiler, where we produce semanticdb. @dos65 did you take a look at that in compiler itself?

**Comment by ritschwumm** [comment]
even worse: ctrl-click is affected, too. i'm learning to click far enough on the left, but it's still a bit annoying.

considering the somewhat longish release cycles of dotty, might it be possible to have a (temporary) workaround in this project?

**Comment by tgodzik** [comment]
This is fixed upstream :tada:

**Comment by tanishiking** [comment]
Confirmed fixed in Scala 3.2.0-RC1 :tada: thank you @dos65 @mirzov 

```scala
trait Stuff[A]:
	def `do`: A
end Stuff

given Stuff[String] with
	val xxx = 1
	def `do`: String =
		xxx + "done" + xxx
```

### 3.1.2
![Screen Shot 2022-06-29 at 14 22 27](https://user-images.githubusercontent.com/9353584/176357996-63c2e20a-5188-44ca-8fbe-f729a725f6c9.png)

### 3.2.0-RC1
![Screen Shot 2022-06-29 at 14 23 10](https://user-images.githubusercontent.com/9353584/176358003-3001ceb2-c797-45d2-b5ed-8e8c6e0d1f95.png)
